### PR TITLE
app-backup/amanda: Clean up ebuild

### PR DESCRIPTION
**Pull request by SoapGentoo**
Closes: https://bugs.gentoo.org/762448
Package-Manager: Portage-3.0.12, Repoman-3.0.2
Signed-off-by: David Seifert <soap@gentoo.org>

### CODE REVIEW

File: app-backup/amanda/amanda-3.5.1-r3.ebuild [hunk 1]
```diff
@@ -86,40 +85,39 @@ amanda_variable_setup() {
...
-	[[ -z "${AMANDA_PORTS_BOTH}" ]] && AMANDA_PORTS_BOTH=
-	[[ -z "${AMANDA_PORTS}" ]] && AMANDA_PORTS=
+	: ${AMANDA_PORTS_UDP:=""}
+	: ${AMANDA_PORTS_TCP:=""}
+	: ${AMANDA_PORTS_BOTH:=""}
+	: ${AMANDA_PORTS:=""}
 
 	# What tar to use
-	[[ -z "${AMANDA_TAR}" ]] && AMANDA_TAR=/bin/tar
+	: ${AMANDA_TAR:=/bin/tar}
```
Comment (thesamesam) [inline app-backup/amanda/amanda-3.5.1-r3.ebuild:103]: ${EPREFIX} here?

File: app-backup/amanda/amanda-3.5.1-r3.ebuild [hunk 2]
```diff
@@ -188,145 +186,170 @@ src_configure() {
...
 	# 2.5.1p2 that it exposes when turned off as well.
-	myconf="${myconf} --with-debugging"
+	myconf+=( --with-debugging )
 	# Where to put our files
-	myconf="${myconf} --localstatedir=${AMANDA_USER_HOMEDIR}"
+	myconf+=( --localstatedir="${AMANDA_USER_HOMEDIR}" )
 
 	# Samba support
-	myconf="${myconf} $(use_with samba smbclient /usr/bin/smbclient)"
+	myconf+=( $(use_with samba smbclient /usr/bin/smbclient) )
```
Comment (thesamesam) [inline app-backup/amanda/amanda-3.5.1-r3.ebuild:243]: I'm not sure we care enough to figure out if the build system prefixes these as needed tbh.

File: app-backup/amanda/amanda-3.5.1-r3.ebuild [hunk 3]
```diff
@@ -403,42 +426,41 @@ src_install() {
...
-	find "${D}"/"${VENDOR_LIB}" -name '*.la' -delete || die
+
+	find "${ED}" -name '*.la' -delete || die
 }
 
 pkg_postinst() {
-	[ ! -f "${TMPENVFILE}" -a "$MERGE_TYPE" == "binary" ] && \
-		TMPENVFILE="${ROOT}${ENVDIR}/${ENVDFILE}"
-	[ ! -f "${TMPENVFILE}" ] && die "Variable setting file (${TMPENVFILE}) should exist!"
+	[[ ! -f ${TMPENVFILE} && ${MERGE_TYPE} == "binary" ]] && TMPENVFILE="${EROOT}${ENVDIR}/${ENVDFILE}"
```
Comment (thesamesam) [inline app-backup/amanda/amanda-3.5.1-r3.ebuild:446]: We get a double slash here but that's not new.

File: app-backup/amanda/amanda-3.5.1-r3.ebuild [hunk 4]
```diff
@@ -188,145 +186,170 @@ src_configure() {
...
+	chmod u=rw,go=r "${root}""${f}" || die
 }
 
 src_install() {
-	[ ! -f "${TMPENVFILE}" ] && die "Variable setting file (${TMPENVFILE}) should exist!"
+	[[ ! -f ${TMPENVFILE} ]] && die "Variable setting file (${TMPENVFILE}) should exist!"
 	source ${TMPENVFILE}
 
 	einfo "Doing stock install"
 	emake DESTDIR="${D}" install
```
Comment (thesamesam) [inline app-backup/amanda/amanda-3.5.1-r3.ebuild:341]: "${ED}" if we are everywhere else?

File: app-backup/amanda/amanda-3.5.1-r3.ebuild [hunk 5]
```diff
@@ -403,42 +426,41 @@ src_install() {
...
 		einfo "A backup is also placed at /etc/amandates.orig"
-		cp -dp "${ROOT}/etc/amandates" "${ROOT}/etc/amandates.orig" || die
-		mkdir -p "${ROOT}/${AMANDA_USER_HOMEDIR}/amanda/" || die
-		cp -dp "${ROOT}/etc/amandates" "${ROOT}/${AMANDA_USER_HOMEDIR}/amanda/amandates" || die
+		cp -dp "${EROOT}"/etc/amandates "${EROOT}"/etc/amandates.orig || die
+		mkdir -p "${EROOT}/${AMANDA_USER_HOMEDIR}/amanda/" || die
+		cp -dp "${EROOT}"/etc/amandates "${EROOT}"/${AMANDA_USER_HOMEDIR}/amanda/amandates || die
 	fi
-	if [ -f "${ROOT}/etc/amandates" ]; then
+	if [[ -f ${EROOT}/etc/amandates ]]; then
```
Comment (thesamesam) [inline app-backup/amanda/amanda-3.5.1-r3.ebuild:458]: "${EROOT}"?

File: app-backup/amanda/amanda-3.5.1-r3.ebuild [hunk 6]
```diff
@@ -403,42 +426,41 @@ src_install() {
...
 		einfo "If you have migrated safely, please delete /etc/amandates"
 	fi
 
 	einfo "Checking setuid permissions"
-	amanda_permissions_fix "${ROOT}"
+	amanda_permissions_fix "${EROOT}"/
 
 	elog "You should configure Amanda in /etc/amanda now."
 	elog
```
Comment (thesamesam) [inline app-backup/amanda/amanda-3.5.1-r3.ebuild:466]: Prefix paths with ${EROOT}? ;)

**Review by thesamesam — COMMENTED** [review]
Tiny comments - none of which are actually problems with what we've changed here.

**Review by thesamesam — APPROVED** [review]

File: app-backup/amanda/amanda-3.5.1-r3.ebuild (hunk 5 above)
Comment (telans) [reply app-backup/amanda/amanda-3.5.1-r3.ebuild:458]: I don't think `[[ ... ]]` does word splitting

Comment (SoapGentoo) [reply app-backup/amanda/amanda-3.5.1-r3.ebuild:458]: correct, which is why I removed the quotes

File: app-backup/amanda/amanda-3.5.1-r3.ebuild (hunk 4 above)
Comment (SoapGentoo) [reply app-backup/amanda/amanda-3.5.1-r3.ebuild:341]: Nope, `${EPREFIX}` has already been added in `econf`

File: app-backup/amanda/amanda-3.5.1-r3.ebuild (hunk 3 above)
Comment (SoapGentoo) [reply app-backup/amanda/amanda-3.5.1-r3.ebuild:446]: nope:
`EROOT` is not slash terminated
`ENVDIR` is not slash terminated
`ENVDFILE` has no leadign slash

File: app-backup/amanda/amanda-3.5.1-r3.ebuild (hunk 2 above)
Comment (SoapGentoo) [reply app-backup/amanda/amanda-3.5.1-r3.ebuild:243]: agreed

**Pull request closed without merging**